plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java

```diff
@@ -279,12 +285,21 @@ protected PinotDataTableWithSize computeNext()
                     return endOfData();
                 }
                 ByteBuffer buffer = response.getPayload().asReadOnlyByteBuffer();
+                DataTable dataTable;
                 try {
-                    return new PinotDataTableWithSize(DataTableFactory.getDataTable(buffer), buffer.remaining());
+                    dataTable = DataTableFactory.getDataTable(buffer);
                 }
                 catch (IOException e) {
                     throw new UncheckedIOException(e);
                 }
+                List<String> exceptions = dataTable.getExceptions().entrySet().stream()
```

Comment:
No need to build `exceptions` object when the exception is an empty. Move to under `if`.